### PR TITLE
Add function to save the solution

### PR DESCRIPTION
**Pull request by hwpang**

### CODE REVIEW

File: src/Simulation.jl [hunk 1]
```diff
@@ -764,3 +764,28 @@ function rates(ssys::Q; ts::X=Array{Float64,1}()) where {Q<:SystemSimulation,X<:
...
+    for (thermovariable, index) in sim.domain.thermovariabledict
+        rename!(df, names(df)[index] => thermovariable)
+    end
+    CSV.write(save_name, df)
+end
+
+function save(syss::T, save_name::String) where {T<:SystemSimulation}
+    df = DataFrame(syss.sol)
+    for sim in syss.sims
+        rename!(df, names(df)[sim.domain.indexes[1]:sim.domain.indexes[2]] .=> sim.names)
```
Comment (mjohnson541) [inline src/Simulation.jl:783]: Is it possible for us to have a name collision if for example you have water in gas phase and liquid phase?

**Review by mjohnson541 — COMMENTED** [review]
Looks mostly good to me. Should we add a read function to go with the save function?

File: src/Simulation.jl (hunk 1 above)
Comment (hwpang) [reply src/Simulation.jl]: Good point. I postfixed the phase name to the species names in the new push.

**Comment by hwpang** [comment]
I rebased with main. @mjohnson541 can you review & merge after the tests pass?

File: src/TestReactors.jl [hunk 2]
```diff
@@ -4,36 +4,36 @@ using Sundials
 
 @testset "Test Reactors" begin
 
-phaseDict = readinput("../src/testing/liquid_phase.rms")
```
Comment (mjohnson541) [inline src/TestReactors.jl:7]: This is going to create nasty merge conflicts with other PRs can we hold off on this until right before v1.0.0?

File: src/Simulation.jl (hunk 1 above)
Comment (mjohnson541) [reply src/Simulation.jl]: Can you squash the fix in here?

**Review by mjohnson541 — CHANGES_REQUESTED** [review]
A couple comments.

**Comment by hwpang** [comment]
> Looks mostly good to me. Should we add a read function to go with the save function?

I don't think it makes sense to add it into this PR, but I agree it would be nice to have a load function. We can open up an issue for that.

**Comment by hwpang** [comment]
@mjohnson541 I addressed the comments!

File: src/TestReactors.jl [hunk 3]
```diff
@@ -608,15 +610,79 @@ end;
 
     inter,pinter = FragmentBasedReactiveFilmGrowthInterfaceConstantT(domainfilm,domainliq,interfacerxns);
 
-    react,y0,p = Reactor((domainfilm,domainliq),(y0film,y0liq),(0.0,0.1),(inter,),(pfilm,pliq,pinter));
```
Comment (mjohnson541) [inline src/TestReactors.jl:611]: What are all these other changes here?

Comment (hwpang) [reply src/TestReactors.jl:611]: Oops probably something went wrong when I rebased this. Will fix it.

Comment (hwpang) [reply src/TestReactors.jl]: Fixed!

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**